### PR TITLE
Rewrite filter unit tests based on criterion

### PR DESCRIPTION
**Pull request by norberttak**
Split and rewrite lib/filter/tests/test_filters.c to be criterion based.
Thanks @Kokan  for the base commits.

### CODE REVIEW

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by kira-syslogng** [comment]
Build SUCCESS

File: lib/filter/tests/test_filters.c [hunk 1]
```diff
@@ -118,31 +118,24 @@ testcase(const gchar *msg,
...
-  static gint testno = 0;
 
-  testno++;
-
-  if (!filter_expr_init(f, configuration))
-    {
-      fprintf(stderr, "Filter init failed; num='%d', msg='%s'\n", testno, msg);
-      exit(1);
-    }
+  filter_expr_init(f, configuration);
```
Comment (furiel) [inline lib/filter/tests/test_filters.c]: The original version asserted for the return value. Please add cr_assert here.

File: lib/filter/tests/test_filters_common.c [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+
+void
+testcase_with_socket(const gchar *msg, const gchar *sockaddr,
+                     FilterExprNode *f,
+                     gboolean expected_result)
+{
+  LogMessage *logmsg;
+  gboolean res;
+
+  filter_expr_init(f, configuration);
```
Comment (furiel) [inline lib/filter/tests/test_filters_common.c]: The original version asserted for the return value. Please add cr_assert here.

File: lib/filter/tests/Makefile.am [hunk 3]
```diff
@@ -1,5 +1,6 @@
 lib_filter_tests_TESTS		 =              \
     lib/filter/tests/test_filters               \
+		lib/filter/tests/test_filters_facility      \
```
Comment (furiel) [inline lib/filter/tests/Makefile.am]: Please unify the whitespaces. I do not have problem with either space or tabs. But with tabs chosen, one tab should be used.

File: lib/filter/tests/Makefile.am [hunk 4]
```diff
@@ -29,6 +30,15 @@ lib_filter_tests_test_filters_facility_SOURCES = 			\
 	lib/filter/tests/test_filters_common.c \
 	lib/filter/tests/test_filters_common.h
 
+lib_filter_tests_test_filters_level_new_CFLAGS     = $(TEST_CFLAGS) \
+	-I${top_srcdir}/lib/filter/tests
+lib_filter_tests_test_filters_level_new_LDADD      = $(TEST_LDADD)  \
+	$(PREOPEN_SYSLOGFORMAT)
+lib_filter_tests_test_filters_level_new_SOURCES = 			\
+	lib/filter/tests/test_filters_level_new.c \
```
Comment (furiel) [inline lib/filter/tests/Makefile.am]: What does the `_new` express in the name?

File: lib/filter/tests/test_filters_regexp.c [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+#include <string.h>
+#include <stdlib.h>
+#include <stdio.h>
+
+TestSuite(filter, .init = setup, .fini = teardown);
+
+typedef struct _FilterParamRegexp
+{
+  const gchar *msg;
+  gint        field;
```
Comment (furiel) [inline lib/filter/tests/test_filters_regexp.c]: We try to avoid such alignment, because when adding new longer variables would cause updating all other unrelated lines.

File: lib/filter/tests/test_filters_fop_cmp.c [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+#include <stdio.h>
+
+TestSuite(filter, .init = setup, .fini = teardown);
+
+typedef struct _FilterParamFopCmp
+{
+  const gchar *msg;
+  const gchar *template1;
+  const gchar *template2;
+  gint        op;
```
Comment (furiel) [inline lib/filter/tests/test_filters_fop_cmp.c]: Unnecessary alignment here too.

File: lib/filter/tests/Makefile.am (hunk 3 above)
Comment (norberttak) [reply lib/filter/tests/Makefile.am]: replaced with TAB

File: lib/filter/tests/Makefile.am (hunk 4 above)
Comment (norberttak) [reply lib/filter/tests/Makefile.am]: the _new comes from filter name filter_level_new, so it's not my name :-)

File: lib/filter/tests/test_filters_regexp.c (hunk 5 above)
Comment (norberttak) [reply lib/filter/tests/test_filters_regexp.c]: whitespace removed

File: lib/filter/tests/test_filters_fop_cmp.c (hunk 6 above)
Comment (norberttak) [reply lib/filter/tests/test_filters_fop_cmp.c]: whitespace removed

File: lib/filter/tests/test_filters_common.c (hunk 2 above)
Comment (norberttak) [reply lib/filter/tests/test_filters_common.c]: cr_assert has been added.

**Comment by kira-syslogng** [comment]
Build SUCCESS
